### ISSUE TITLE
samgeo.hq_sam not working in docker 

### ISSUE THREAD

**Issue by glemoine62** [opengeos/segment-geospatial#223]
Same issue as reported in: https://github.com/opengeos/segment-geospatial/issues/173

However, since the solution provided (install with ```pip install git+https://github.com/SysCV/sam-hq.git```) there does not work, I tried running the docker installation instead. But throws same error: 

```RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.```

suggesting the docker image was not built with the alternative pip install (or, indeed, the alternative install does not work).

### Environment Information

-   samgeo version: docker: giswqs/segment-geospatial:latest
-   Python version: 3.11.6
-   Operating System: linux

**Comment by MarcioDDCF** [comment]
Is this related to [https://github.com/opengeos/segment-geospatial/issues/201] ?

**Comment by glemoine62** [comment]
No. Not related to #201. I want to run samgeo.hq_sam on the CPU in the same way you can run "normal" samgeo on the CPU. 
The solution proposed in #173 does not work, whether you install locally or in a docker container. 

The speed difference between running inference with samgeo on a GPU or a CPU is not too dramatic, esp. when testing (GPU order of 6 faster). The models require a pretty large GPU forcing you to move to cloud resources. I prefer running on CPU.

**Comment by flxcp** [comment]
Try adding `device='cpu'` when initializing SamGeo:

```python
from samgeo.hq_sam import SamGeo

sam = SamGeo(model_type="vit_h", sam_kwargs=None, device='cpu')
```

See if it helps!

**Comment by glemoine62** [comment]
Thanks for the tip, but I tried that, and it doesn't work. The problem is in the sam_hq implementation, which does not check for the device key. To get this solved, you need to go into /opt/conda/lib/python3.11/site-packages/segment_anything_hq/build_sam.py

and edit line 160:

```
state_dict = torch.load(f, map_location=torch.device('cpu'))
```

which is for the vit_h model, you need to edit the equivalent statements for the other models. This is the quick and dirty solution, it would be better to do this on the basis of the device keyword.

(I assume use of the giswqs/segment-geospatial docker image. I commit the revised container to giswqs/segment-geospatial-cpu)

The CPU inference is some factor 6 slower (on 1024x1024 tiles) than on a high-end hosted GPU, but that's good enough for testing.
